### PR TITLE
Profile/aw/79307 save direct deposit pt3 the testinging

### PR DESCRIPTION
**Pull request by adamwhitlock1**
## Summary

- Adds e2e tests for most happy paths in Direct Deposit
- fixes PUT requests by making sure the request body only includes the 3 required fields, but most importantly also makes sure the header are set for accept json/application

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/79307

## Testing done

- Lots of e2e tests added

## Screenshots

No UI changes, just tests and one api call update

## What areas of the site does it impact?

Profile -> direct deposit

## Acceptance criteria

- [x] Tests added
- [x] PUT requests have correct headers set

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed

### CODE REVIEW

File: src/applications/personalization/profile/actions/directDeposit.js [hunk 1]
```diff
@@ -98,16 +98,56 @@ export function saveDirectDeposit(
 
     const response = await getData(DIRECT_DEPOSIT_API_ENDPOINT, {
       method: 'PUT',
+      headers: {
+        'Content-Type': 'application/json',
+      },
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/actions/directDeposit.js:103]: @tpharrison here is where I'm fixing my PUT request headers. They will merge with the default headers of the defaultSettings:

```
const defaultSettings = {
    method: 'GET',
    credentials: 'include',
    headers: {
      'X-Key-Inflection': 'camel',
      'Source-App-Name': window.appName,
      'X-CSRF-Token': csrfTokenStored,
    },
  };
  ```

File: src/applications/personalization/profile/hooks/useDirectDeposit.js [hunk 2]
```diff
@@ -102,7 +102,11 @@ export const useDirectDeposit = () => {
   const onFormSubmit = useCallback(
     () => {
       const payload = {
-        paymentAccount: formData,
+        paymentAccount: {
+          accountType: formData.accountType,
+          accountNumber: formData.accountNumber,
+          routingNumber: formData.routingNumber,
+        },
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/hooks/useDirectDeposit.js:109]: Here I figured it is probably best to pluck out just the 3 required fields to send in the PUT body

File: src/applications/personalization/profile/actions/directDeposit.js [hunk 3]
```diff
@@ -98,16 +98,56 @@ export function saveDirectDeposit(
...
     });
 
-    // error handling for the response
-    if (response.error || response.errors) {
-      recordDirectDepositEvent({
+    // for client side errors
+    // check if response is an instance of an error TypeError or any other error
+    if (response instanceof Error) {
+      const event = {
+        method: 'PUT',
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/actions/directDeposit.js:111]: Added some client side error handling for if the user loses internet or another client side error occurs.

File: src/applications/personalization/profile/actions/directDeposit.js [hunk 4]
```diff
@@ -98,16 +98,56 @@ export function saveDirectDeposit(
...
 
+      return;
+    }
+
+    // error handling for the response if it is a server side error
+    if (response.error || response.errors) {
+      const errorStatus =
+        response.errors?.[0]?.status || response?.status || '';
+      const errorCodeOrMessage =
+        response.errors?.[0]?.code || response?.error || 'unknown-error';
```
Comment (adamwhitlock1) [inline src/applications/personalization/profile/actions/directDeposit.js:139]: Parsing the error code or the error message depending on if it is an errors array or an error string (500 errors are a string for instance) I also thought tagging on the status code would be useful for GA as well, and that will just be appended to the end of the 'error-key' that we send GA

**Review by tpharrison — APPROVED** [review]
LGTM 🚀

**Review by tpharrison — APPROVED** [review]

**Pull request merged**